lib/log.bash

```diff
@@ -55,6 +55,6 @@ function _log_error()
   example '$ _log_error "Failed to load git plugin..."'
   group 'log'
 
-  [[ "$BASH_IT_LOG_LEVEL" -ge $BASH_IT_LOG_LEVEL_ERROR ]] || return 0
-  _log_general "${echo_red}" "ERROR: " "$1"
+  [[ "${BASH_IT_LOG_LEVEL:-1}" -ge $BASH_IT_LOG_LEVEL_ERROR ]] || return 0
```

Comment:
as `1` is `BASH_IT_LOG_LEVEL_ERROR` I suggest replacing it with `0` here

Reply:
Yeah, I went around several times not sure what to default to. Error and warning should print by default, I think. 

Right now, there is no default and the test fails for bad syntax if the variable isn't set (when `set -u`), so there's no Pre existing behavior to maintain.

Reply:
The test want to assure that nothing is printed if the variable is not defined. I think the correct solution would be to set this to 0 if it is not initialized (we normally do not log at all, only on `bash-it doctor` call)